packages/mc-html-template/lib/load-env.js

```diff
@@ -1,16 +1,53 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
 const fs = require('fs');
 
 // Keep a reference to the loaded config so that requiring the module
 // again will result in returning the cached value.
 let loadedEnv;
 
-module.exports = configPath => {
-  if (loadedEnv) return loadedEnv;
+const START_OF_ENV_PLACEHODER = 'env:';
+const isEnvVariablesPlaceholder = valueOfEnvConfig =>
+  valueOfEnvConfig && valueOfEnvConfig.startsWith(START_OF_ENV_PLACEHODER);
+const substitutePlaceholder = valueOfEnvConfig => {
+  const nameOfEnvVariable = valueOfEnvConfig.replace(
+    START_OF_ENV_PLACEHODER,
+    ''
+  );
+
+  const valueOfEnvVariable = process.env[nameOfEnvVariable];
+
+  if (!valueOfEnvVariable) {
+    throw new Error(
+      `Missing '${nameOfEnvVariable}' specified in config as 'env:${nameOfEnvVariable}'.`
+    );
+  }
+
+  return valueOfEnvVariable;
+};
+const substituteEnvVariablePlaceholders = config => {
+  const entriesOfEnvConfig = Object.entries(config);
+
+  const replacedEntriesOfEnvConfig = entriesOfEnvConfig.map(
+    ([envConfigKey, envConfigValue]) => [
+      envConfigKey,
+      isEnvVariablesPlaceholder(envConfigValue)
+        ? substitutePlaceholder(envConfigValue)
+        : envConfigValue,
+    ]
+  );
+
+  return Object.fromEntries(replacedEntriesOfEnvConfig);
+};
+
+module.exports = (configPath, { disableCache } = { disableCache: false }) => {
+  if (loadedEnv && !disableCache) return loadedEnv;
 
   // Attempt to read the JSON file provided by each application
   const rawConfig = fs.readFileSync(configPath, { encoding: 'utf8' });
   const config = JSON.parse(rawConfig);
 
+  const substitutedConfig = substituteEnvVariablePlaceholders(config);
```

Comment:
Nit: maybe "compile" instead of "substitute"? We also use that name for "compiling" the `index.html`

Reply:
Compiling to me is something different than string replacing. When going around the internet on this most people call it substitution.

Reply:
Usually when you have template strings or something you compile the template. See e.g. ejs, handlebars, etc.

Anyway, not a big deal. I'm also fine with this.